packages/commonwealth/client/scripts/state/api/reactionCounts/fetchReactionCounts.ts

```diff
@@ -5,15 +5,25 @@ import app from 'state';
 
 // const THREAD_REACTION_COUNTS_STALE_TIME = 30 * 1_000; // 30 s
 
-interface FetchThreadReactionCountsProps {
+interface FetchReactionCountsProps {
   // chainId: string; // this should be used when we fully migrate thread controller to react query
-  threadIds: number[];
+  address: string;
+  threadIds?: number[];
+  proposalIds?: string[];
+  commentIds?: string[];
 }
 
-const fetchThreadReactionCounts = async ({ threadIds }: FetchThreadReactionCountsProps) => {
+const fetchReactionCounts = async ({
+  address,
+  threadIds,
+  proposalIds,
+  commentIds
+}: FetchReactionCountsProps) => {
   const response = await axios.post(`${app.serverUrl()}/reactionsCounts`, {
-    thread_ids: threadIds,
-    active_address: app.user.activeAccount?.address,
+    ...(threadIds && threadIds.length > 0 && { thread_ids: threadIds }),
```

Comment:
could be 
`...(threadIds?.length > 0 && { thread_ids: threadIds }),`

Reply:
Yeah, that looks cleaner 👍. Done

---
Also on a similar note, there is this weird thing with the `/reactionsCounts` API, it can take the proposal, comment, and thread ids and return respective reactions.

- In [master--ViewThreadPage.tsx#L257-L266](https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/pages/view_thread/ViewThreadPage.tsx#L257-L266) we are passing comment and proposal ids to get reaction counts
- And in [master--controllers/server/threads.ts#L775-L782](https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/controllers/server/threads.ts#L775-L782) we are passing thread ids to the same endpoint

I don't know if there is a planned refactor to split the reactionCounts API into comments, threads, and proposal APIs.

cc: @rbennettcw @jnaviask

Reply:
I agree that it would make sense (in terms of REST) to split it out:
- `/threads/:id/reactions`
- `/comments/:id/reactions`
etc.

We've already done so with the routes responsible for *creating* reactions.

I can create a ticket for this.

Reply:
@rbennettcw can you link me the ticket?

Reply:
@jnaviask here's the ticket: https://github.com/hicommonwealth/commonwealth/issues/4575